components/cldr-json-data-provider/src/transform/dates.rs

```diff
@@ -0,0 +1,368 @@
+        ])
+    }
+}
+
+impl From<&cldr_json::StylePatterns> for gregory::patterns::StylePatternsV1 {
+    fn from(other: &cldr_json::StylePatterns) -> Self {
+        Self {
+            full: other.full.clone(),
+            long: other.long.clone(),
+            medium: other.medium.clone(),
+            short: other.short.clone(),
+        }
+    }
+}
+
+impl From<&cldr_json::Dates> for gregory::DatesV1 {
+    fn from(other: &cldr_json::Dates) -> Self {
+        Self {
+            symbols: Some(gregory::DateSymbolsV1 {
+                months: other.calendars.gregorian.months.as_ref().map(|months| months.into()),
+                weekdays: other.calendars.gregorian.days.as_ref().map(|days| days.into()),
+                ..Default::default()
+            }),
+            patterns: Some(gregory::PatternsV1 {
+                date: other.calendars.gregorian.date_formats.as_ref().map(|formats| formats.into()),
+                time: other.calendars.gregorian.time_formats.as_ref().map(|formats| formats.into()),
+                date_time: other.calendars.gregorian.date_time_formats.as_ref().map(|formats| formats.into()),
+                ..Default::default()
+            }),
+        }
+    }
+}
+
+impl From<&cldr_json::months::Contexts> for gregory::months::ContextsV1 {
+    fn from(other: &cldr_json::months::Contexts) -> Self {
+        Self {
+            format: other.format.as_ref().map(|format| format.into()),
+            stand_alone: other.stand_alone.as_ref().map(|format| format.into()),
+        }
+    }
+}
+
+impl From<&cldr_json::months::Widths> for gregory::months::WidthsV1 {
+    fn from(other: &cldr_json::months::Widths) -> Self {
+        Self {
+            abbreviated: other.abbreviated.as_ref().map(|width| width.into()),
+            narrow: other.narrow.as_ref().map(|width| width.into()),
+            short: other.short.as_ref().map(|width| width.into()),
+            wide: other.wide.as_ref().map(|width| width.into()),
+        }
+    }
+}
+
+impl From<&cldr_json::days::Contexts> for gregory::weekdays::ContextsV1 {
+    fn from(other: &cldr_json::days::Contexts) -> Self {
+        Self {
+            format: other.format.as_ref().map(|format| format.into()),
+            stand_alone: other.stand_alone.as_ref().map(|format| format.into()),
+        }
+    }
+}
+
+impl From<&cldr_json::days::Widths> for gregory::weekdays::WidthsV1 {
+    fn from(other: &cldr_json::days::Widths) -> Self {
+        Self {
+            abbreviated: other.abbreviated.as_ref().map(|width| width.into()),
+            narrow: other.narrow.as_ref().map(|width| width.into()),
+            short: other.short.as_ref().map(|width| width.into()),
+            wide: other.wide.as_ref().map(|width| width.into()),
+        }
+    }
+}
+
+/// Serde structs for the CLDR JSON dates files.
+pub(self) mod cldr_json {
+    use crate::cldr_langid::CldrLangID;
+    use serde::Deserialize;
+    use std::borrow::Cow;
+
+    macro_rules! symbols {
```

Comment:
Optional: Since you are only using this macro 2 times, it might be more readable if you expanded it.